src/shogun/lib/observers/ParameterObserver.h

```diff
@@ -64,13 +36,30 @@ namespace shogun
 		 */
 		ParameterObserver(std::vector<std::string>& parameters);
 
+		/**
+		 * Constructor
+		 * @param parameters list of parameters which we want to watch over
+		 */
+		ParameterObserver(std::vector<ParameterProperties>& properties);
```

Comment:
@karlnapf @vigsterkr since these observers will be mainly used by the users, do we still need these constructors or would it be better to populate the member variables by using only `put` calls?

Reply:
fine from my side, but no hard feelings. I think Viktor might prefer c++ ctors to remain in there (reason the lib might be used natively)

Reply:
Okay, I will keep them in case they need to be used natively.